Shared/Samples/Edit feature attachments/EditFeatureAttachmentsView.swift

```diff
@@ -0,0 +1,232 @@
+                        try await model.setSelectedFeature(for: feature)
+                        if let location = mapProxy.location(fromScreenPoint: point) {
+                            model.updateCalloutPlacement(to: location)
+                        }
+                    } catch {
+                        self.error = error
+                    }
+                }
+                .overlay(alignment: .center) {
+                    if !isLoaded {
+                        ProgressView("Loading…")
+                            .padding()
+                            .background(.ultraThinMaterial)
+                            .cornerRadius(10)
+                            .shadow(radius: 50)
+                    }
+                }
+        }
+        .errorAlert(presentingError: $error)
+    }
+}
+
+private extension EditFeatureAttachmentsView {
+    // MARK: - AttachmentSheetView
+    
+    struct AttachmentSheetView: View {
+        /// The error shown in the error alert.
+        @State private var error: Error?
+        
+        /// The data model for the sample.
+        @ObservedObject var model: Model
+        
+        var body: some View {
+            VStack {
+                Spacer()
+                List {
+                    ForEach(0...$model.attachments.count, id: \.self) { index in
+                        if index == model.attachments.count {
+                            AddAttachmentView(onAdd: {
+                                Task {
+                                    do {
+                                        guard let pngData = UIImage(
+                                            named: "PinBlueStar"
+                                        )?.pngData() else {
+                                            return
+                                        }
+                                        try await model.add(
+                                            name: "Attachment",
+                                            type: "png",
+                                            dataElement: pngData
+                                        )
+                                    } catch {
+                                        self.error = error
+                                    }
+                                }
+                            })
+                        } else {
+                            let attachment = model.attachments[index]
+                            AttachmentView(attachment: attachment, onDelete: { attachment in
+                                Task {
+                                    do {
+                                        try await model.delete(attachment: attachment)
+                                    } catch {
+                                        self.error = error
+                                    }
+                                }
+                            })
+                        }
+                    }
+                }
+            }
```

Comment:
```suggestion
            Form {
                Section {
                    List {
                        ForEach($model.attachments.indices, id: \.self) { index in
                            let attachment = model.attachments[index]
                            AttachmentView(attachment: attachment, onDelete: { attachment in
                                Task {
                                    do {
                                        try await model.delete(attachment: attachment)
                                    } catch {
                                        self.error = error
                                    }
                                }
                            })
                        }
                    }
                }
                Section {
                    AddAttachmentView(onAdd: {
                        Task {
                            do {
                                guard let pngData = UIImage(
                                    named: "PinBlueStar"
                                )?.pngData() else {
                                    return
                                }
                                try await model.add(
                                    name: "Attachment",
                                    type: "png",
                                    dataElement: pngData
                                )
                            } catch {
                                self.error = error
                            }
                        }
                    })
                }
            }
```
A Form with sections should be used instead of a List which requires the index to be checked to determine if the `AddAttachmentView` should be shown or not. 

<img src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/117859673/ecd224c5-3553-4f80-9f51-4a21116318df" width="300">